guides/09_other-tutorials/wrapping-layouts.md

```diff
@@ -0,0 +1,538 @@
+# Wrapping Layouts
+
+Tags: LAYOUTS
+
+## Introduction
+
+Gradio features [blocks](https://www.gradio.app/docs/blocks) to easily layout applications. To use this feature, you need to stack or nest layout components and create a hierarchy with them. This implementation isn't difficult to implement and maintain for small projects, but after the project gets complex, this component hierarchy becomes difficult to maintain and reusing layouts becomes difficult.
+
+In this guide, we are going to explore how we can wrap the layout classes to create more maintainable and easy-to-read applications without sacrificing flexibility.
+
+## Implementation
+
+- The wrapping utility has two important classes. The first one is the ```LayoutBase``` class and the other one is the ```Application``` class. Let's start with these classes' implementations first. Then we look at how we can wrap the layouts from the gradio package with the ```LayoutBase``` class.
+
+### LayoutBase Class
+
+- This class is going to be the base class for the layout classes we are going to wrap. For example; for ```gradio.layouts.Row``` layout, we are going to wrap it with the `RowLayout` class which is inherited from the ```LayoutBase``` class.
+- The `LayoutBase` class has four variables. Let's write them and the implementation of the constructor:
+ 
+```python
+class LayoutBase:
+    main_layout: Block
+    name: str
+    global_children_dict: Dict[str, Block]
+    renderables: list
+
+    def __init__(self) -> None:
+        self.main_layout = None
+        self.name = "Layout Base"
+        self.global_children_dict = {}
+        self.renderables = []
+```
+
+- Here are the variables' definitions:
+
+|       Variable       |              Type              | Definition                                                                                                                                        |
+| :------------------: | :----------------------------: | :------------------------------------------------------------------------------------------------------------------------------------------------ |
+|     main_layout      |      gradio.blocks.Block       | Stores the main layout from the gradio package of this class. For example, storing ```gradio.layouts.Row``` layout for the ```RowLayout``` class. |
+|         name         |              str               | Name of the class. Using to differentiate from other layouts and for debug purposes.                                                              |
+| global_children_dict | Dict[str, gradio.blocks.Block] | Stores the children components with given name.                                                                                                   |
+|     renderables      |              list              | Stores the renderable elements such as components and layouts.                                                                                    |
+
+---------------------------------------------
```

Comment:
Why are you using this line breaks everywhere? They don't look so good when they're rendered.

Reply:
Maybe if you want to seperate the sections you can use numbers for the big headings and bullet points for the smaller ones?